src/backend/joanie/core/api/admin.py

```diff
@@ -221,6 +221,8 @@ def create(self, request, *args, **kwargs):
         data = request.data
         serializer = self.get_serializer(data=data)
         data["product"] = kwargs.get("product_id")
+        if data.get("course_runs", None) == "":
+            data["course_runs"] = []
         if not serializer.is_valid():
             return Response(serializer.errors, status=400)
         course_runs = serializer.validated_data.pop("course_runs", [])
```

Comment:
Instead of the modification that you proposed, this would do the same and come after form validation?
```suggestion
        course_runs = serializer.validated_data.pop("course_runs", []) or []
```

Reply:
It's the validation that is an issue here, when the front does not send a course_run, it sends {"course_run": ""}, it being an empty string does not match the array expected

Reply:
IC